lib/src/request/request.rs

```diff
@@ -616,6 +616,12 @@ impl<'r> Request<'r> {
         *self.state.params.borrow_mut() = route.get_param_indexes(self.uri());
     }
 
+    /// Set the method of `self`, even if self is a shared reference
+    #[inline(always)]
+    pub(crate) fn set_method_interior(&self, method: Method) {
```

Comment:
Let's name this `_set_method()`. We should use `_method_name()` when we have an internal-only method that conflicts with a public method.